### PR TITLE
Deny users without roles

### PR DESCRIPTION
**Pull request by gijskant**

### CODE REVIEW

File: transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * This filter denies access to users without any roles assigned to them.
+ */
+@Slf4j
+class TransmartKeycloakAuthenticationProcessingFilter extends KeycloakAuthenticationProcessingFilter {
+
+    TransmartKeycloakAuthenticationProcessingFilter(AuthenticationManager authenticationManager) {
+        super(authenticationManager)
+        log.info("Enable filter that denies access to users without any roles assigned to them.")
```
Comment (ewelinagr) [inline transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy]: other token-verification logs (e.g. from the keycloak-client plugin) are on the error level, maybe useful to have this one as error too

File: transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) throws AuthenticationException, IOException, ServletException {
+        Authentication authentication = super.attemptAuthentication(request, response)
+        if (authentication && authentication.authorities.empty) {
```
Comment (ewelinagr) [inline transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy]: works fine, but grants access to any role, also an invalid one (other than ROLE_ADMIN, ROLE_PUBLIC and transmart-format specific)

File: transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy (hunk 1 above)
Comment (gijskant) [reply transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy]: But error level should only be used for errors, right? these are not errors.

File: transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy (hunk 2 above)
Comment (gijskant) [reply transmart-api-server/src/main/groovy/org/transmartproject/api/server/server/TransmartKeycloakAuthenticationProcessingFilter.groovy]: The authentication validation now parses the authorities to TranSMART access levels.

**Pull request merged**